core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessSendRetryManager.java

```diff
@@ -0,0 +1,51 @@
+package org.eclipse.dataspaceconnector.transfer.core.transfer;
+
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.retry.WaitStrategy;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+
+import java.time.Clock;
+import java.util.function.Supplier;
+
+import static java.lang.String.format;
+
+/**
+ * Service enabling a "long retry" mechanism when sending {@link TransferProcess}es across applications.
+ * The implementation supports a pluggable retry strategy (e.g. an exponential wait mechanism
+ * so as not to overflow the remote service when it becomes available again).
+ */
+public class TransferProcessSendRetryManager implements SendRetryManager<TransferProcess> {
+    private final Monitor monitor;
+    private final Supplier<WaitStrategy> delayStrategySupplier;
+    private final int retryLimit;
+    protected Clock clock = Clock.systemUTC();
+
+    public TransferProcessSendRetryManager(Monitor monitor, Supplier<WaitStrategy> delayStrategySupplier, int retryLimit) {
+        this.monitor = monitor;
+        this.delayStrategySupplier = delayStrategySupplier;
+        this.retryLimit = retryLimit;
+    }
+
+    @Override
+    public boolean shouldDelay(TransferProcess process) {
+        int retryCount = process.getStateCount() - 1;
+        if (retryCount <= 0) {
+            return false;
+        }
+        var delayStrategy = delayStrategySupplier.get();
+        delayStrategy.failures(retryCount);
```

Comment:
The amount of failures are added to `errorCount` within the `failures` method here, but as far as I can tell the state count is increased as well every time a request fails. Are we counting things twice?

Reply:
Added comments to make the use of `Supplier` more visible